src/pages/api/multipleProductValidity.ts

```diff
@@ -0,0 +1,130 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import Cookies from 'cookies';
+import {
+    NUMBER_OF_PRODUCTS_COOKIE,
+    MULTIPLE_PRODUCT_COOKIE,
+    OPERATOR_COOKIE,
+    CSV_ZONE_UPLOAD_COOKIE,
+    PERIOD_SINGLE_OPERATOR_SERVICES_COOKIE,
+    PERIOD_TYPE_COOKIE,
+    MATCHING_DATA_BUCKET_NAME,
+} from '../../constants/index';
+import { isSessionValid } from './service/validator';
+import { redirectToError, setCookieOnResponseObject, getDomain, redirectTo } from './apiUtils';
+import { Product } from '../multipleProductValidity';
+import { getCsvZoneUploadData, putStringInS3 } from '../../data/s3';
+import { batchGetStopsByAtcoCode, Stop } from '../../data/auroradb';
+
+interface DecisionData {
+    operatorName: string;
+    type: string;
+    nocCode: string;
+    products: Product[];
+    selectedServices?: [];
+    zoneName?: string;
+    stops?: Stop[];
+}
+
+export const addErrorsIfInvalid = (req: NextApiRequest, numberOfProducts: number, products: Product[]): Product[] => {
+    const limiter = new Array(numberOfProducts);
+    const response: Product[] = [];
+    for (let i = 0; i < limiter.length; i += 1) {
+        let validity = req.body[`validity-row${i}`];
+        let error = '';
+        if (!validity) {
+            validity = '';
+            error = 'Select one of the two validity options';
+        }
+        const product = {
+            productName: products[i].productName,
+            productNameId: products[i].productNameId,
+            productPrice: products[i].productPrice,
+            productPriceId: products[i].productPriceId,
+            productDuration: products[i].productDuration,
+            productDurationId: products[i].productDurationId,
+            productValidity: { validity, error },
+        };
+        response.push(product);
+    }
+    return response;
+};
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        if (!isSessionValid(req, res)) {
+            throw new Error('Session is invalid.');
+        }
+        const cookies = new Cookies(req, res);
+        const operatorCookie = unescape(decodeURI(cookies.get(OPERATOR_COOKIE) || ''));
```

Comment:
We created a helper function to do the unescape and decode in a recent ticket, it would be worth using that